delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/client/ElasticSearchClient.scala

```diff
@@ -558,4 +563,47 @@ object ElasticSearchClient {
     implicit val decodeCount: Decoder[Count] =
       Decoder.instance(_.get[Long]("count").map(Count(_)))
   }
+
+  sealed trait BulkResponse
+  object BulkResponse {
+    final case object NoErrors                                extends BulkResponse
+    final case class SomeErrors(items: List[BulkItemOutcome]) extends BulkResponse
+  }
+
+  sealed trait BulkItemOutcome
+  object BulkItemOutcome {
+    final case object Success                extends BulkItemOutcome
+    final case class Error(json: JsonObject) extends BulkItemOutcome
+  }
+
+  implicit private val bulkItemOutcomeDecoder = Decoder.instance[BulkItemOutcome] { hcursor =>
+    val obj = hcursor
+      .downField("update")
+      .success
+      .orElse(hcursor.downField("create").success)
+      .orElse(hcursor.downField("delete").success)
+      .orElse(hcursor.downField("index").success)
+      .toRight(DecodingFailure("expected item of type update, create, delete or index", hcursor.history))
+    obj
+      .map(itemCursor =>
+        itemCursor.get[JsonObject]("error") match {
+          case Left(_)      => BulkItemOutcome.Success
+          case Right(value) => BulkItemOutcome.Error(value)
+        }
+      )
+  }
+
+  object BatchPartialFailure {
```

Comment:
Could this go in the companion object of `BulkErrors` (or whatever it is called) ?